s3-source-connector/src/integration-test/java/io/aiven/kafka/connect/s3/source/IntegrationBase.java

```diff
@@ -0,0 +1,211 @@
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.AdminClientConfig;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.serialization.ByteArrayDeserializer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.connect.json.JsonDeserializer;
+
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.client.builder.AwsClientBuilder;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.github.dockerjava.api.model.Ulimit;
+import io.confluent.kafka.serializers.KafkaAvroDeserializer;
+import org.apache.avro.generic.GenericRecord;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.TestInfo;
+import org.testcontainers.containers.Container;
+import org.testcontainers.containers.KafkaContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.localstack.LocalStackContainer;
+import org.testcontainers.utility.DockerImageName;
+
+public interface IntegrationBase {
+
+    String DOCKER_IMAGE_KAFKA = "confluentinc/cp-kafka:7.7.0";
+
+    default AdminClient newAdminClient(final KafkaContainer kafka) {
+        final Properties adminClientConfig = new Properties();
+        adminClientConfig.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, kafka.getBootstrapServers());
+        return AdminClient.create(adminClientConfig);
+    }
+
+    default ConnectRunner newConnectRunner(final KafkaContainer kafka, final File pluginDir,
+            final int offsetFlushIntervalMs) {
+        return new ConnectRunner(pluginDir, kafka.getBootstrapServers(), offsetFlushIntervalMs);
+    }
+
+    static void extractConnectorPlugin(File pluginDir) throws IOException, InterruptedException {
+        final File distFile = new File(System.getProperty("integration-test.distribution.file.path"));
+        assert distFile.exists();
+
+        final String cmd = String.format("tar -xf %s --strip-components=1 -C %s", distFile, pluginDir.toString());
+        final Process process = Runtime.getRuntime().exec(cmd);
+        assert process.waitFor() == 0;
+    }
+
+    static File getPluginDir() throws IOException {
+        final File testDir = Files.createTempDirectory("s3-source-connector-for-apache-kafka-test-").toFile();
+
+        final File pluginDir = new File(testDir, "plugins/s3-source-connector-for-apache-kafka/");
```

Comment:
This "plugins/.." should be a static class variable same for "s3-source-connector..." above

Reply:
Not sure why it should be a static class variable. This is an interface, and declaring those modifiers are redundant.

Reply:
I missed that this was an interface, however it is good practice to have class constants instead of strings defined, static final is best practice for those constants as you don't normally want them changed, final is unrequired here though as you mentioned because it is an interface.

Reply:
agree, its best practice. but it is redundant, as they are by default static